### PR TITLE
Issue #8206: Resolve PMD warning deprecated attribute 'ClassOrInterfaceDeclaration/@Image

### LINKED ISSUE #8206: Resolve PMD warning deprecated attribute 'ClassOrInterfaceDeclaration/@Image'
PMD issues a warnings for each suppression:
```
[WARNING] Use of deprecated attribute 'ClassOrInterfaceDeclaration/@Image' in a findChildNodesWithXPath navigation, please use @SimpleName instead
java.lang.RuntimeException: Use of deprecated attribute 'ClassOrInterfaceDeclaration/@Image' in a findChildNodesWithXPath navigation, please use @SimpleName instead
```

PMD configuration files should be updated to illuminate these warnings.

### PR DESCRIPTION
**Pull request by Gaurav-Punjabi**
Fixes #8206 

#### 1. Solution
Needed to replace `@Image` with `@SimpleName` as `@Image` is deprecated now.

### CODE REVIEW

**Comment by Gaurav-Punjabi** [comment]
@pbludov @romani @strkkk Please review.

**Comment by rnveach** [comment]
> Please review.
>  Gaurav-Punjabi marked this pull request as draft

@Gaurav-Punjabi This PR is listed as "Draft" and that usually means it is not finalized and probably not ready for a review. Is this a draft because you plan to make more changes?

**Comment by Gaurav-Punjabi** [comment]
@rnveach Yeah, actually I noticed later that more changes need to be done. I'm working on them. Will mark this as ready and ping you once it's done.

**Comment by Gaurav-Punjabi** [comment]
@rnveach It's ready for review now. Please check

File: config/pmd.xml [hunk 1]
```diff
@@ -354,8 +361,8 @@
       overloads a synchronized method, so it should have synchronized
       modifier. -->
       <property name="violationSuppressXPath"
-        value="//ClassOrInterfaceDeclaration[@Image='UniqueProperties'
-          or @Image='SequencedProperties']//MethodDeclaration[@Name='keys' or @Name='put']"/>
+        value="//ClassOrInterfaceDeclaration[@SimpleName='UniqueProperties'
```
[the remark this answers is not in the document]
Comment (Gaurav-Punjabi) [reply config/pmd.xml]: Done

File: config/pmd.xml [hunk 2]
```diff
@@ -339,8 +345,9 @@
            Checker collects external resource locations and sets up the configuration.
            CheckstyleAntTask integrates Checkstyle with Ant.
             -->
-      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='Main'
-        or @Image='Checker' or @Image='CheckstyleAntTask']"/>
+      <property name="violationSuppressXPath"
+        value="//ClassOrInterfaceDeclaration[@SimpleName='Main'
```
[the remark this answers is not in the document]
Comment (Gaurav-Punjabi) [reply config/pmd.xml]: Done

File: config/pmd.xml [hunk 3]
```diff
@@ -312,8 +317,8 @@
       <!-- The class PackageObjectFactory has many boilerplate methods.
            JavadocMethodCheck is in the process of being rewritten. -->
       <property name="violationSuppressXPath"
-                value="//ClassOrInterfaceDeclaration[@Image='PackageObjectFactory']
-          |//ClassOrInterfaceDeclaration[@Image='JavadocMethodCheck']"/>
+                value="//ClassOrInterfaceDeclaration[@SimpleName='PackageObjectFactory']
+          |//ClassOrInterfaceDeclaration[@SimpleName='JavadocMethodCheck']"/>
```
[the remark this answers is not in the document]
Comment (Gaurav-Punjabi) [reply config/pmd.xml]: Done

File: config/pmd.xml [hunk 4]
```diff
@@ -268,23 +271,25 @@
...
            SuppressFilterElement is a single constructor and can't be split easily -->
-      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='Main'
-        or @Image='PackageObjectFactory' or @Image='RequireThisCheck'
-        or @Image='VariableDeclarationUsageDistanceCheck'
-        or @Image='HandlerFactory']
-      | //ClassOrInterfaceDeclaration[@Image='XMLLogger']//MethodDeclaration[@Name='encode']
-      | //ClassOrInterfaceDeclaration[@Image='LeftCurlyCheck' or @Image='FinalLocalVariableCheck'
-        or @Image='NPathComplexityCheck']
+      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[
+        @SimpleName='Main'
```
[the remark this answers is not in the document]
Comment (Gaurav-Punjabi) [reply config/pmd.xml]: Done

File: config/pmd.xml [hunk 5]
```diff
@@ -194,7 +196,8 @@
     <properties>
       <!-- There is no alternative to using System.exit in the CLI class
            to report the return code. -->
-      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='Main']"/>
+      <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[
```
[the remark this answers is not in the document]
Comment (Gaurav-Punjabi) [reply config/pmd.xml]: Done

File: config/pmd.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gaurav-Punjabi) [reply config/pmd.xml]: Done

**Pull request merged**